x-pack/functionbeat/provider/aws/cli_manager.go

```diff
@@ -0,0 +1,336 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package aws
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/external"
+	"github.com/awslabs/goformation/cloudformation"
+	merrors "github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/x-pack/functionbeat/core"
+	"github.com/elastic/beats/x-pack/functionbeat/provider"
+)
+
+const (
+	// AWS lambda currently support go 1.x as a runtime.
+	runtime     = "go1.x"
+	bucket      = "functionbeat-deploy"
```

Comment:
This cannot be a const, bucket names are shared by all S3 users so we need to generate a unique name here

Reply:
Instead of making it random and having to either delete it or save the state on disk I've made it a setting of the aws provider.